decentraland/unity-explorer
# fix: preview not working during festival

- Refactor on how we set the start position with possible overrides
- Removes the logic from `RealmNavigator`
- Takes into consideration a position set by args so the feature flag is ignore accordingly
- Also sets the correct value for `localhost` in `Hostname` for feature flag usage.


Fixes: https://github.com/decentraland/creator-hub/issues/312

### QA TEST STEPS

- Start the client with the flags `--debug --dclenv zone`. You should start in 100,100
- Start the client with the flags `--debug --dclenv zone --position 150,150`. You should start in 150,150
- Start the client with the flags `--debug --dclenv zone --realm olavra.dcl.eth`. Dont mind that the scene doesnt load, just check that the coordinates are `0,0`. (Also, move a little bit through the scene, probably the scene will say `-150,-150`. If you move a parcel it should change correclty)

Local scene

1. Download [this test scene](https://github.com/user-attachments/files/17851248/quick-00-scene-test.zip) and uncompress it somewhere
2. Enter the scene root folder with a terminal/command and run `npm i` and then `npm run start`
3. Close the Explorer that auto-opened. Leave the scene running in the console/terminal.
4. Download the build from this PR and connect it to the running scene using [the console command](https://github.com/decentraland/unity-explorer/wiki/How-to-connect-to-a-local-scene-(Unity-Editor---Custom-Build---Latest-Released-Build)#connecting-a-custom-build-to-the-scene) according to your OS. Add the `--dclenv zone` as well to the args
5. The client should open and the scene with the cube in the middle should be shown
6. Close the client
7. Close the terminal/command running the scene
8. Open the client from this PR again
9. It should connect to 100,100
10. Teleport to a world, for example `/goto olavra` (Same issue, it wont load, but doesnt matter)
11. Check the world loads correctly
12. Press the button "Go back to Genesis City"
13. It should take you to 100,100 (During the festival the avatar should spawn in the festival coordinates)
